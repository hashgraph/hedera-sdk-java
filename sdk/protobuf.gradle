dependencies {
    implementation 'io.grpc:grpc-protobuf:1.50.2'

    // Protobuf Lite is used to maintain easy compatibility with Android
    // https://github.com/protocolbuffers/protobuf/blob/master/java/lite.md
    api "com.google.protobuf:protobuf-java:3.21.8"
}

protobuf {
    generateProtoTasks {
        all()*.plugins {
            grpc {}
        }
    }
}
