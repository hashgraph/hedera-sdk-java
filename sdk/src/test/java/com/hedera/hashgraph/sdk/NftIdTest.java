package com.hedera.hashgraph.sdk;

import com.google.protobuf.InvalidProtocolBufferException;
import io.github.jsonSnapshot.SnapshotMatcher;
import org.bouncycastle.util.encoders.Hex;
import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.util.concurrent.TimeoutException;

class NftIdTest {

    static Client mainnetClient = null;
    static Client testnetClient = null;
    static Client previewnetClient = null;

    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
        mainnetClient = Client.forMainnet();
        testnetClient = Client.forTestnet();
        previewnetClient = Client.forPreviewnet();
    }

    @AfterClass
    public static void afterAll() throws TimeoutException {
        mainnetClient.close();
        testnetClient.close();
        previewnetClient.close();
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    void fromString() {
        SnapshotMatcher.expect(NftId.fromString("0.0.5005@1234").toString()).toMatchSnapshot();
    }

    @Test
    void fromStringWithChecksumOnMainnet() {
        SnapshotMatcher.expect(NftId.fromString("0.0.123-vfmkw@7584").toStringWithChecksum(mainnetClient)).toMatchSnapshot();
    }

    @Test
    void fromStringWithChecksumOnTestnet() {
        SnapshotMatcher.expect(NftId.fromString("0.0.123-rmkyk@584903").toStringWithChecksum(testnetClient)).toMatchSnapshot();
    }

    @Test
    void fromStringWithChecksumOnPreviewnet() {
        SnapshotMatcher.expect(NftId.fromString("0.0.123-ntjly@487302").toStringWithChecksum(previewnetClient)).toMatchSnapshot();
    }

    @Test
    void toBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(Hex.toHexString(new TokenId(5005).nft(4920).toBytes())).toMatchSnapshot();
    }

    @Test
    void fromBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(NftId.fromBytes(new TokenId(5005).nft(574489).toBytes()).toString()).toMatchSnapshot();
    }
}
