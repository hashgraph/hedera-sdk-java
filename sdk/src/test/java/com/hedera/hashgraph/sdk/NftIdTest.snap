com.hedera.hashgraph.sdk.NftIdTest.fromBytes=[
  "0.0.5005/574489"
]


com.hedera.hashgraph.sdk.NftIdTest.fromString=[
  "0.0.5005/1234"
]


com.hedera.hashgraph.sdk.NftIdTest.fromStringWithChecksumOnMainnet=[
  "0.0.123-vfmkw/7584"
]


com.hedera.hashgraph.sdk.NftIdTest.fromStringWithChecksumOnPreviewnet=[
  "0.0.123-ntjly/487302"
]


com.hedera.hashgraph.sdk.NftIdTest.fromStringWithChecksumOnTestnet=[
  "0.0.123-rmkyk/584903"
]


com.hedera.hashgraph.sdk.NftIdTest.toBytes=[
  "0a03188d2710b826"
]